angular-client/src/pages/charging-page/components/fault-display/fault-display.component.ts

```diff
@@ -19,33 +20,29 @@ export default class FaultDisplay {
 
   ngOnInit() {
     this.storage.get(IdentifierDataType.COMM_TIMEOUT_FAULT).subscribe((value) => {
-      if (value.values[0]) {
-        this.faults.push({ faultName: 'Communication Timeout', time: new Date().toLocaleTimeString() });
-      }
+      this.faultPushCheck(floatPipe(value.values[0]));
     });
 
     this.storage.get(IdentifierDataType.HARDWARE_FAILURE_FAULT).subscribe((value) => {
-      if (value.values[0]) {
-        this.faults.push({ faultName: 'Hardware Failure', time: new Date().toLocaleTimeString() });
-      }
+      this.faultPushCheck(floatPipe(value.values[0]));
     });
 
     this.storage.get(IdentifierDataType.OVER_TEMP_FAULT).subscribe((value) => {
-      if (value.values[0]) {
-        this.faults.push({ faultName: 'Over Temp', time: new Date().toLocaleTimeString() });
-      }
+      this.faultPushCheck(floatPipe(value.values[0]));
     });
 
     this.storage.get(IdentifierDataType.VOLTAGE_WRONG_FAULT).subscribe((value) => {
-      if (value.values[0]) {
-        this.faults.push({ faultName: 'Voltage Wrong', time: new Date().toLocaleTimeString() });
-      }
+      this.faultPushCheck(floatPipe(value.values[0]));
     });
 
     this.storage.get(IdentifierDataType.WRONG_BAT_CONNECT_FAULT).subscribe((value) => {
-      if (value.values[0]) {
-        this.faults.push({ faultName: 'Wrong Battery Connect', time: new Date().toLocaleTimeString() });
-      }
+      this.faultPushCheck(floatPipe(value.values[0]));
     });
   }
+
+  faultPushCheck(numberValue: number) {
+    if (numberValue !== 0 && this.faults.length <= 50) {
```

Comment:
the second half of this is bad because it's gonna throw out any new fault messages once we hit 50. I want it to do the opposite. We should have a queue of faults where when we hit 50 messages the oldest fault is popped off the queue so the newest one can be added to the front